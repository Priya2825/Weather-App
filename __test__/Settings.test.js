import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import Setting from "../src/screens/Settings";
import { NavigationContainer } from "@react-navigation/native";

jest.mock("@react-native-async-storage/async-storage", () => ({
  getItem: jest.fn(),
  setItem: jest.fn(),
}));

describe("Setting Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with given data", () => {
    const tree = render(
      <NavigationContainer>
        <Setting />
      </NavigationContainer>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("loads the notification setting from AsyncStorage", async () => {
    // Mock AsyncStorage value (notification is enabled)
    AsyncStorage.getItem.mockResolvedValue("true");

    const { getByTestId } = render(<Setting />);

    // Wait for async data to load and the state to be updated
    await waitFor(() => {
      const switchElement = getByTestId("switch");
      expect(switchElement.props.value).toBe(true);
    });
  });

  it("updates AsyncStorage when the switch is toggled on", async () => {
    AsyncStorage.getItem.mockResolvedValue("false");

    const { getByTestId } = render(<Setting />);
    const switchElement = getByTestId("switch");

    // Simulate toggling the switch to 'on'
    fireEvent(switchElement, "valueChange", true);

    // Check that the switch is enabled
    expect(switchElement.props.value).toBe(true);

    // Wait for AsyncStorage.setItem to be called with the correct value
    await waitFor(() => {
      expect(AsyncStorage.setItem).toHaveBeenCalledWith(
        "notification-enabled",
        "true"
      );
    });
  });
});
