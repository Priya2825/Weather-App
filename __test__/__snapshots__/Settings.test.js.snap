// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting Component renders correctly with given data 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text>
      Enable Notifications
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#81b0ff"
      style={
        [
          {
            "height": 31,
            "width": 51,
          },
          {
            "backgroundColor": "#3e3e3e",
            "borderRadius": 16,
          },
        ]
      }
      testID="switch"
      thumbTintColor="#f4f3f4"
      tintColor="#767577"
      value={false}
    />
  </View>
  <Text />
</View>
`;
